app/containers/App/index.js

```diff
@@ -24,24 +26,26 @@ const AppWrapper = styled.div`
   flex-direction: column;
 `;
 
-export function App(props) {
-  return (
-    <AppWrapper>
-      <Helmet
-        titleTemplate="%s - React.js Boilerplate"
-        defaultTitle="React.js Boilerplate"
-      >
-        <meta name="description" content="A React.js Boilerplate application" />
-      </Helmet>
-      <Header />
-      {React.Children.toArray(props.children)}
-      <Footer />
-    </AppWrapper>
-  );
+export class App extends React.Component { // eslint-disable-line react/prefer-stateless-function
+  render() {
+    return (
+      <AppWrapper>
+        <Helmet
+          titleTemplate="%s - React.js Boilerplate"
+          defaultTitle="React.js Boilerplate"
+        >
+          <meta name="description" content="A React.js Boilerplate application" />
+        </Helmet>
+        <Header />
+        <Switch>
```

Comment:
I think we're almost ready here!

@Dattaya Want to confirm one point - in my previous version, I had made `Routes` a connected component and passed the store's `location` in here as a prop. I was having trouble with reliable route-rerendering without making sure the `Switch` uses our redux state rather than its own. After the change, all rerendering issues I found went away - FWIW the issues would only show up with a production build on a real server, not local with webpack dev server.

Do you have any specific reason not to propagate `location` into here from the store? If not I can make the change to restore the connection.